docker/dockerfiles/allin1/entrypoint.sh

```diff
@@ -35,6 +35,44 @@ setup_priority_networks()
     echo "priority_networks = 127.0.0.1/32" >> $SR_HOME/be/conf/be.conf
 }
 
+log_stderr()
+{
+    echo "`date --rfc-3339=seconds` $@" >&2
+}
+
+logerror()
+{
+    log_stderr "ERROR $@"
+}
+
+update_fe_conf_if_run_in_shared_data_mode()
+{
+    if [ ! -f $SR_HOME/fe/meta/image/VERSION ]; then
+        run_mode="$RUN_MODE"
+        run_mode=${run_mode,,} # to lower case
+        if [ "$run_mode" == "shared_data" ]; then # if run mode is shared_data, we need to update fe.conf
+            echo "# config for shared_data mode
+run_mode = shared_data
+cloud_native_meta_port = 6090
+# Whether volume can be created from conf. If it is enabled, a builtin storage volume may be created.
+enable_load_volume_from_conf = false" >> $SR_HOME/fe/conf/fe.conf
+        fi
+    else
+        # If $SR_HOME/fe/meta/image/VERSION file exists, it should have a line like "runMode=shared_nothing" or "runMode=shared_data"
+        previous_run_mode=`cat $SR_HOME/fe/meta/image/VERSION | grep runMode | cut -d '=' -f 2`
+        previous_run_mode=${previous_run_mode,,} # to lower case
+
+        run_mode="$RUN_MODE"
+        run_mode=${run_mode,,} # to lower case
+
+        # check if previous_run_mode and run_mode are the same
+        if [ -n "$run_mode" ] && [ "$previous_run_mode" != "$run_mode" ]; then
+            logerror "Error: previous_run_mode in $SR_HOME/fe/meta/image/VERSION and RUN_MODE environment must be the same"
```

Comment:
log the detected value, let the user know exactly what the mismatch is.

Reply:
starting a allin1 container is very easy, and I think we can let user to fix the mismatch.

Reply:
`restart` doesn't help, it has to delete the container and recreate it. Anyway, it is good enough now.